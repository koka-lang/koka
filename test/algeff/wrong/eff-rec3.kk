 
effect state<a> {
  fun get() : a
  fun set(x : a) : ()
}

val new-ref = handler(s) {
  get() -> resume(s,s)
  set(x) -> resume((),x)
}

fun recurse() {
  using new-ref(id)
  fun foo(i : int) {
    get()(i)
  }
  set(foo)
  foo(42)
}
