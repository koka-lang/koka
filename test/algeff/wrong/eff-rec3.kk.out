test/algeff/wrong/eff-rec3.kk(15, 5): error: types do not match (due to an infinite type)
 context : get()(i)
 term : get()
 inferred effect: <state<(int) -> _e1 _a>|_e>
 expected effect: _e1
 because : effect cannot be subsumed
 hint : annotate the function definition?